mobile/spec/group.spec.ts

```diff
@@ -0,0 +1,302 @@
+        const user5 = new KeyPair();
+
+        // creation of the first group contract
+        let variables: IGroupDefinition = {
+            orgPubKeys: [user1._public.toHex(), user2._public.toHex(), user3._public.toHex(), user4._public.toHex(), user5._public.toHex()],
+            purpose: "Test",
+            suite: "edwards25519",
+            voteThreshold: ">=1/3",
+        };
+
+        // creates five group contracts
+        //              c0
+        //              |
+        //              c1
+        //            /    \
+        //           c2    c3
+        //            \    /
+        //              c4
+
+        const contractCollection = new GroupContractCollection(variables.purpose);
+        let gd = new GroupDefinition(variables);
+        const c0 = contractCollection.createGroupContract(undefined, gd);
+        variables = gd.allVariables;
+        variables.predecessor = [c0.id];
+        const c1 = contractCollection.createGroupContract(c0, new GroupDefinition(c0.groupDefinition.allVariables));
+        contractCollection.sign(c1, user1._private);
+        contractCollection.sign(c1, user2._private);
+        contractCollection.sign(c1, user3._private);
+        contractCollection.sign(c1, user4._private);
+        contractCollection.sign(c1, user5._private);
+
+        variables = gd.allVariables;
+        variables.voteThreshold = ">=1/2";
+        variables.predecessor = [c1.id];
+        gd = new GroupDefinition(variables);
+        const c2 = contractCollection.createGroupContract(c1, gd);
+        contractCollection.sign(c2, user1._private);
+        contractCollection.sign(c2, user2._private);
+        contractCollection.sign(c2, user3._private);
+
+        variables = gd.allVariables;
+        variables.voteThreshold = ">=9/10";
+        gd = new GroupDefinition(variables);
+        const c3 = contractCollection.createGroupContract(c1, gd);
+        contractCollection.sign(c3, user4._private);
+        contractCollection.sign(c3, user5._private);
+
+        variables = gd.allVariables;
+        variables.voteThreshold = ">=1/2";
+        variables.predecessor = [c2.id, c3.id];
+        gd = new GroupDefinition(variables);
+        const c4 = contractCollection.createGroupContract(c2, gd);
+        contractCollection.sign(c4, user1._private);
+        contractCollection.sign(c4, user2._private);
+        contractCollection.sign(c4, user3._private);
+        contractCollection.sign(c4, user4._private);
+        contractCollection.sign(c4, user5._private);
+
+        expect(contractCollection.isAccepted(c1)).toBeTruthy();
+        expect(contractCollection.isAccepted(c2)).toBeTruthy();
+        expect(contractCollection.isAccepted(c3)).toBeTruthy();
+        expect(contractCollection.isAccepted(c4)).toBeTruthy();
+        expect(contractCollection.getCurrentGroupContract()).toEqual(c4);
+        Log.print("Test merge passed!");
+    });
+});
+
+// // helping method
+function sleep(milliseconds) {
```

Comment:
not used - can be removed. And a better way to sleep is this:

```
function delay(ms: number) {
    return new Promise( resolve => setTimeout(resolve, ms) );
}

await delay(100);
```

Reply:
I've removed it